app/src/main/java/com/example/encoratask/vm/BaseViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.encoratask.vm
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.encoratask.model.Character
+import com.example.encoratask.repository.APICharacterImpl
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class BaseViewModel : ViewModel() {
+
+    @Inject
+    lateinit var api: APICharacterImpl
+
+    private val _listCharacter by lazy {
+        MutableLiveData<List<Character>>()
+    }
+
+    val listCharacter: LiveData<List<Character>> = _listCharacter
```

Comment:
variable listCharacter is not used, right? I can notice you're doing everything on _listCharacter